packages/token/index.ts

```diff
@@ -1,18 +1,98 @@
-import type { HexString } from "~typing"
-
-export class Token {
-  public constructor(
-    public readonly address: HexString,
-    public readonly symbol: string,
-    public readonly decimals: number
-  ) {}
+import { Contract, Interface } from "ethers"
+
+import type { ContractRunner } from "~packages/node"
+import type { BigNumberish, HexString } from "~typing"
+
+export type Token = {
+  address: HexString
+  symbol: string
+  decimals: number
+}
+
+export const ETH: Token = {
+  address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+  symbol: "ETH",
+  decimals: 18
 }
```

Comment:
As I see it, `packages/token/index.ts` is mainly for ERC20 token. 
It's a little weird that we put constant ETH information here (but we can think about it later).

Reply:
From my understanding, in the extension, even native ETH is considered a token.
We might need a way to identify if the user has chosen ETH or another ERC20 token.

Reply:
> From my understanding, in the extension, even native ETH is considered a token.

: I think the same as you, perhaps we can keep ETH constant here and move ERC20 related code into `packages/contract/erc20.ts`